### ISSUE TITLE
binary operation `==` cannot be applied to type `std::string::FromUtf8Error` upon install

### ISSUE THREAD

**Issue by kvz** [jameslittle230/stork#45]
Hi there, cool project you got going! Wanted to kick its tires but got this:

```bash
$ cargo install stork-search
    Updating crates.io index
  Downloaded stork-search v0.7.1
  Downloaded 1 crate (150.7 KB) in 1.54s
  Installing stork-search v0.7.1
  Downloaded wasm-bindgen v0.2.63
  Downloaded console_error_panic_hook v0.1.6
  Downloaded htmlescape v0.3.1
  Downloaded num-format v0.4.0
  Downloaded rust-stemmers v1.2.0
  Downloaded arrayvec v0.4.12
  Downloaded wasm-bindgen-macro v0.2.63
  Downloaded srtparse v0.2.0
  Downloaded nodrop v0.1.14
  Downloaded wasm-bindgen-macro-support v0.2.63
  Downloaded rmp-serde v0.14.3
  Downloaded wasm-bindgen-backend v0.2.63
  Downloaded wasm-bindgen-shared v0.2.63
  Downloaded bumpalo v3.4.0
  Downloaded rmp v0.8.9
   Compiling proc-macro2 v1.0.18
   Compiling unicode-xid v0.2.0
   Compiling syn v1.0.31
   Compiling serde v1.0.112
   Compiling wasm-bindgen-shared v0.2.63
   Compiling log v0.4.8
   Compiling cfg-if v0.1.10
   Compiling byteorder v1.3.4
   Compiling ryu v1.0.5
   Compiling bumpalo v3.4.0
   Compiling autocfg v1.0.0
   Compiling lazy_static v1.4.0
   Compiling itoa v0.4.5
   Compiling serde_json v1.0.55
   Compiling wasm-bindgen v0.2.63
   Compiling arrayvec v0.4.12
   Compiling nodrop v0.1.14
   Compiling srtparse v0.2.0
   Compiling htmlescape v0.3.1
   Compiling quote v1.0.7
   Compiling num-traits v0.2.12
   Compiling num-format v0.4.0
   Compiling rmp v0.8.9
   Compiling wasm-bindgen-backend v0.2.63
   Compiling wasm-bindgen-macro-support v0.2.63
   Compiling serde_derive v1.0.112
   Compiling wasm-bindgen-macro v0.2.63
   Compiling bincode v1.2.1
   Compiling toml v0.5.6
   Compiling rust-stemmers v1.2.0
   Compiling rmp-serde v0.14.3
   Compiling console_error_panic_hook v0.1.6
   Compiling stork-search v0.7.1
error[E0369]: binary operation `==` cannot be applied to type `std::string::FromUtf8Error`
  --> /home/kvz/.cargo/registry/src/github.com-1ecc6299db9ec823/stork-search-0.7.1/src/searcher/index_analyzer.rs:64:28
   |
64 |     VersionStringUtf8Error(std::string::FromUtf8Error),
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: an implementation of `std::cmp::PartialEq` might be missing for `std::string::FromUtf8Error`

error[E0369]: binary operation `!=` cannot be applied to type `std::string::FromUtf8Error`
  --> /home/kvz/.cargo/registry/src/github.com-1ecc6299db9ec823/stork-search-0.7.1/src/searcher/index_analyzer.rs:64:28
   |
64 |     VersionStringUtf8Error(std::string::FromUtf8Error),
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: an implementation of `std::cmp::PartialEq` might be missing for `std::string::FromUtf8Error`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0369`.
error: failed to compile `stork-search v0.7.1`, intermediate artifacts can be found at `/tmp/cargo-installzmiYls`

Caused by:
  could not compile `stork-search`.

To learn more, run the command again with --verbose.
```

Does that ring any bell?

**Comment by jameslittle230** [comment]
No, thanks for letting me know! I'll look into it and check back.

**Comment by jameslittle230** [comment]
Hey, thanks for being patient.

I was surprised to learn that `FromUtf8Error` [only started implementing `PartialEq` five months ago](https://github.com/rust-lang/rust/commit/847d5b4d1387a30f1798a5c3c59c3e0c31e00319). I'd be curious to know if you're using a Rust version from >5 months ago -- if so, you might have to run `rustup update` to get Stork to build and run.

To make sure nothing was going wrong, I nuked Rust from my system, reinstalled, and confirmed that with a clean install of Rust 1.44.0, Stork can build successfully. 

Thanks for checking out Stork! I'm sorry you had trouble before you even got it working.

**Comment by lukethacoder** [comment]
Ran into the same issue with version `rustc 1.40.0 (73528e339 2019-12-16)`.

running `rustup update` did nothing.

try setting the default rustup toolchain. `rustup default INSERT_TOOLCHAIN_HERE`.
Running the above returned this "info" message:
```
info: note that the toolchain '1.40.0-i686-pc-windows-msvc' is currently in use (directory override for '\\?\C:\Users\USER_DIR')
```

Running `rustup override set stable-i686-pc-windows-msvc` solved my issues and successfully updated me to `rustc 1.44.1 (c7087fe00 2020-06-17)` (latest rust version)

---

**TL:DR**
`rustup override set stable-i686-pc-windows-msvc` (or the current stable version)

https://github.com/rust-lang/rustup/issues/451 may also help

**Comment by jameslittle230** [comment]
ah, thanks, @lukethacoder. Sorry you hit that, but glad you were able to figure out how to resolve it.

I'll leave this issue open for a bit more time for visibility, but will close it in a week or so.
